account_id.go

```diff
@@ -3,30 +3,57 @@ package hedera
 import (
 	"fmt"
 	protobuf "github.com/golang/protobuf/proto"
-	"strings"
-
 	"github.com/hashgraph/hedera-sdk-go/v2/proto"
+	"github.com/pkg/errors"
+	"strings"
 )
 
 // AccountID is the ID for a Hedera account
 type AccountID struct {
-	Shard   uint64
-	Realm   uint64
-	Account uint64
+	Shard    uint64
+	Realm    uint64
+	Account  uint64
+	Checksum *string
+	Network  *NetworkName
 }
 
 // AccountIDFromString constructs an AccountID from a string formatted as
 // `Shard.Realm.Account` (for example "0.0.3")
-func AccountIDFromString(s string) (AccountID, error) {
-	shard, realm, num, err := idFromString(s)
+func AccountIDFromString(data string) (AccountID, error) {
+	var checksum parseAddressResult
+	var err error
+
+	var networkNames = []NetworkName{
+		Mainnet,
+		Testnet,
+		Previewnet,
+	}
+
+	var network NetworkName
+	for _, name := range networkNames {
+		checksum, err = checksumParseAddress(name.Network(), data)
+		if err != nil {
```

Comment:
You're not supposed to err here. You're supposed to try the next network name. If all 3 fail, then you err.

Actually, maybe I'm misunderstanding. Not sure what's the difference between err and `checksum.status`